### PR TITLE
OS loader Optimization for Raw Partition Boot.

### PR DESCRIPTION
**Pull request by naveenm18**
Removed Checking for MBR/GPT partitions.
In FindPartitions() identifying memory device
using MediaGetInterfaceType().

Signed-off-by: Mutha <naga.naveen.mutha@intel.com>

### CODE REVIEW

File: BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c [hunk 1]
```diff
@@ -542,7 +540,7 @@ FindPartitions (
 
   CopyMem (&PartBlockDev->BlockInfo, &DevBlockInfo, sizeof (DEVICE_BLOCK_INFO));
 
-  if (DevBlockInfo.BlockSize == 1) {
+  if (MediaGetInterfaceType() == OsBootDeviceMemory) {
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:543]: For both SPI and memory device.

File: BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c [hunk 2]
```diff
@@ -551,36 +549,18 @@ FindPartitions (
...
-      // Could not find any partition, so assume no partitions
-      BlockDev              = & (PartBlockDev->BlockDevice[0]);
-      BlockDev->StartBlock  = 0;
-      BlockDev->LastBlock   = DevBlockInfo.BlockNum - 1;
-      PartBlockDev->PartitionType    = EnumPartTypeUnknown;
-      PartBlockDev->PartitionChecked = TRUE;
-      PartBlockDev->BlockDeviceCount = 1;
-      Status = EFI_SUCCESS;
-    }
+  // Could not find any partition, so assume no partitions
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:552]: This is not correct.  Why did you remove all the partition check ?
It is still required for other devices, such as USB, SATA, UFS, etc.  It is a common function for all.  We just need to handle SPI/Mem device specially.  Remaining logic should be kept untouched.

File: BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c [hunk 3]
```diff
@@ -551,36 +549,18 @@ FindPartitions (
...
 
-    DEBUG ((DEBUG_INFO, "Partition type: %a  (%d logical partitions)\n", \
-            GetPartitionTypeName (PartBlockDev->PartitionType), \
-            PartBlockDev->BlockDeviceCount));
-  } else {
-    Status = EFI_DEVICE_ERROR;
-  }
+  DEBUG ((DEBUG_INFO, "Partition type: %a  (%d logical partitions)\n", \
+          GetPartitionTypeName (PartBlockDev->PartitionType), \
+          PartBlockDev->BlockDeviceCount));
```
Comment (myguitar) [inline BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:563]: Agree with Maurice. This common logic should be kept.

File: BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c (hunk 2 above)
Comment (naveenm18) [reply BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:552]: Ok. I considered only SPI/MEM. I will revert this change because if SPI/MEM it is handled earlier.

File: BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c [hunk 4]
```diff
@@ -522,6 +522,7 @@ FindPartitions (
   LOGICAL_BLOCK_DEVICE        *BlockDev;
   EFI_PARTITION_TABLE_HEADER  *Gpt;
   PART_BLOCK_DEVICE           *PartBlockDev;
+  OS_BOOT_MEDIUM_TYPE          mCurrentMediaType;
```
Comment (myguitar) [inline BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:525]: Can you remove 'm' prefix?

File: BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c [hunk 5]
```diff
@@ -532,6 +533,8 @@ FindPartitions (
     return Status;
   }
 
+  mCurrentMediaType = MediaGetInterfaceType();
```
Comment (myguitar) [inline BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:536]: Can you move this line to right before using the variable?

File: BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c [hunk 6]
```diff
@@ -542,7 +545,7 @@ FindPartitions (
 
   CopyMem (&PartBlockDev->BlockInfo, &DevBlockInfo, sizeof (DEVICE_BLOCK_INFO));
 
-  if (DevBlockInfo.BlockSize == 1) {
+  if (mCurrentMediaType == OsBootDeviceMemory || mCurrentMediaType == OsBootDeviceMemory) {
```
Comment (myguitar) [inline BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:548]: I guess you meant OsBootDeviceSpi.
```
if (mCurrentMediaType == OsBootDeviceMemory || mCurrentMediaType == OsBootDeviceMemory)
->
if ((CurrentMediaType == OsBootDeviceSpi) || (CurrentMediaType == OsBootDeviceMemory))
```

Quick question. Did the original code 'if (DevBlockInfo.BlockSize == 1)' cause a problem? Do we really need this change?

Comment (mauricema) [reply BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:548]: Using BlockSize==1 to check SPI/Mem device will work as long as you set block size to 1. But for SPI, we originally have bigger block size.  I think it is always good to use the actual media type for check.

Comment (gdong1) [reply BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:548]: OsBootDeviceSpi is mainly used for PDR region. For Memory mapped SPI BIOS region, OsBootDeviceMemory should be used for better performance. What issue for this patch is to solve?

Comment (naveenm18) [reply BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:548]: OsBootDeviceSpi is supported for BIOS region also now. This patch if for optimization and improving the code checks.

Comment (myguitar) [reply BootloaderCommonPkg/Library/PartitionLib/PartitionLib.c:548]: Understood. So, this is code improvement to check actual media type. Please address review comments.
